### PR TITLE
chore: first piece of the merge for allowing vscode-util in faux package Part 1

### PR DESCRIPTION
**Pull request by gbockus-sf**
### What does this PR do?
package and tsconfig changes for making it possible to have normal imports from vscode-utils.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,9 +17,10 @@
     "@commitlint/config-conventional": "^17.0.3",
     "@monorepo-utils/workspaces-to-typescript-project-references": "2.8.2",
     "@octokit/core": "4.0.5",
-    "@salesforce/dev-config": "1.6.0",
+    "@salesforce/dev-config": "^3.1.0",
```
Comment (gbockus-sf) [inline package.json:20]: This required with the typescript version bump.  Only fallout was they have removed the tslint configuration so I had to copy it over to our base configuration file.

File: package.json [hunk 2]
```diff
@@ -38,13 +40,15 @@
     "shelljs": "0.8.5",
     "shx": "0.3.4",
     "snyk": "^1.364.0",
-    "ts-loader": "5.4.5",
+    "ts-jest": "28.0.8",
```
Comment (gbockus-sf) [inline package.json:43]: used for unit testing

File: package.json [hunk 3]
```diff
@@ -38,13 +40,15 @@
...
+    "ts-node": "10.9.1",
     "tslint": "5.5.0",
-    "typescript": "3.8.3",
+    "typescript": "^4.7.4",
     "vsce": "1.103.1",
     "vscode-test": "1.6.1",
-    "webpack": "4.30.0",
-    "webpack-cli": "3.3.12"
+    "webpack": "^5.74.0",
+    "webpack-cli": "^4.10.0"
```
Comment (gbockus-sf) [inline package.json:51]: Note this versions should be consistent across all the packages.

File: packages/salesforcedx-apex-debugger/package.json [hunk 4]
```diff
@@ -12,7 +12,7 @@
     "Debuggers"
   ],
   "dependencies": {
-    "@salesforce/salesforcedx-utils-vscode": "55.11.0",
+    "@salesforce/salesforcedx-utils": "55.11.0",
```
Comment (gbockus-sf) [inline packages/salesforcedx-apex-debugger/package.json:15]: debuggers (not the debugger adapters) should depend on utils not vscode-utils.

File: packages/salesforcedx-apex-debugger/package.json [hunk 5]
```diff
@@ -47,8 +51,8 @@
     "watch": "tsc -watch -p .",
     "clean": "shx rm -rf node_modules && shx rm -rf out && shx rm -rf coverage && shx rm -rf .nyc_output",
     "test": "npm run test:unit && npm run test:integration",
-    "test:unit": "node ./node_modules/nyc/bin/nyc.js ./node_modules/mocha/bin/_mocha --recursive out/test/unit --exit --reporter mocha-multi-reporters --reporter-options configFile=../../config/mochaUnitTestsConfig.json",
-    "test:integration": "node ./node_modules/cross-env/dist/bin/cross-env.js VSCODE_NLS_CONFIG={} ./node_modules/nyc/bin/nyc.js ./node_modules/mocha/bin/_mocha --recursive out/test/integration --reporter mocha-multi-reporters --reporter-options configFile=../../config/mochaIntegrationTestsConfig.json"
+    "test:unit": "jest --coverage",
+    "test:integration": "jest --coverage -c ./jest.integration.config.js"
```
Comment (gbockus-sf) [inline packages/salesforcedx-apex-debugger/package.json:55]: This is the standard jest config for all projects. If a project is using the vscode test running for integration tests that was left in place.

File: packages/salesforcedx-sobjects-faux-generator/package.json [hunk 6]
```diff
@@ -8,39 +8,45 @@
...
+    "ts-jest": "28.0.8",
     "ts-node": "^9.1.1",
-    "typescript": "3.8.3",
-    "webpack": "5.68.0"
+    "typescript": "^4.7.4",
+    "webpack": "^5.74.0"
   },
   "scripts": {
-    "compile": "webpack",
+    "compile": "tsc -p ./ && webpack",
```
Comment (gbockus-sf) [inline packages/salesforcedx-sobjects-faux-generator/package.json:41]: standardize this.  Don't depend on webpack for typescript compiles.

File: packages/salesforcedx-vscode-apex-replay-debugger/package.json [hunk 7]
```diff
@@ -28,6 +28,7 @@
     "@salesforce/core": "^3.23.2",
     "@salesforce/salesforcedx-apex-replay-debugger": "55.11.0",
     "@salesforce/salesforcedx-utils-vscode": "55.11.0",
+    "@salesforce/salesforcedx-utils": "55.11.0",
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-apex-replay-debugger/package.json:31]: Note this is the debugger adapter module so it pulling both vscode-utils and utils.

File: tsconfig.common.json [hunk 8]
```diff
@@ -20,6 +20,8 @@
     "declarationMap": true,
     "preserveConstEnums": true,
     "noImplicitAny": true,
+    // TODO: cleanup unknown usage in catch blocks and remove below setting
+    "useUnknownInCatchVariables": false
```
Comment (gbockus-sf) [inline tsconfig.common.json:24]: the typescript update included no longer just letting untyped errors in catch blocks.  I started fixing these but there were a ton so I ended up just putting this place.

File: packages/salesforcedx-apex-replay-debugger/package.json [hunk 9]
```diff
@@ -41,8 +44,8 @@
     "watch": "tsc -watch -p .",
     "clean": "shx rm -rf node_modules && shx rm -rf out && shx rm -rf coverage && shx rm -rf .nyc_output",
     "test": "npm run test:unit && npm run test:integration",
-    "test:unit": "node ./node_modules/nyc/bin/nyc.js ./node_modules/mocha/bin/_mocha --recursive out/test/unit --reporter mocha-multi-reporters --reporter-options configFile=../../config/mochaUnitTestsConfig.json",
-    "test:integration": "node ./node_modules/cross-env/dist/bin/cross-env.js VSCODE_NLS_CONFIG={} ./node_modules/nyc/bin/nyc.js ./node_modules/mocha/bin/_mocha --recursive out/test/integration --reporter mocha-multi-reporters --reporter-options configFile=../../config/mochaIntegrationTestsConfig.json"
+    "test:unit": "jest --coverage",
```
Comment (RitamAgrawal) [inline packages/salesforcedx-apex-replay-debugger/package.json:47]: Will it provide test coverage for only jest files?

File: packages/salesforcedx-vscode-apex-replay-debugger/package.json (hunk 7 above)
Comment (RitamAgrawal) [reply packages/salesforcedx-vscode-apex-replay-debugger/package.json:31]: How do we know which is the adapter module and not? is the -replay-debugger adapter module?

File: tsconfig.common.json (hunk 8 above)
Comment (RitamAgrawal) [reply tsconfig.common.json:24]: So after cleaning up unknown usage, we change the value to true or remove the "useUnknownInCatchVariables" altogether?

File: packages/salesforcedx-apex-debugger/package.json (hunk 5 above)
Comment (RitamAgrawal) [reply packages/salesforcedx-apex-debugger/package.json:55]: jest and --coverage configurations are defined in jest.base.config file, which is not included in this PR. It's a part of PR part-3 4429.

File: packages/salesforcedx-apex-replay-debugger/package.json (hunk 9 above)
Comment (RitamAgrawal) [reply packages/salesforcedx-apex-replay-debugger/package.json:47]: That's great! So all the tests will be covered in this command, mocha as well as jest.

File: package.json (hunk 3 above)
Comment (RitamAgrawal) [reply package.json:51]: Have verified these version bumps across the packages. it's consistent.

File: packages/salesforcedx-vscode-apex-replay-debugger/package.json (hunk 7 above)
Comment (gbockus-sf) [reply packages/salesforcedx-vscode-apex-replay-debugger/package.json:31]: sadly today by naming convention or by reading the config in the package.json files.  For us the salesforcedx-apex-debugger and salesforcedx-apex-replay-debugger are our debuggers and the salesforcedx-vscode-apex-debugger and salesforcedx-vscode-apex-replay-debugger are the adapters.

File: tsconfig.common.json (hunk 8 above)
Comment (gbockus-sf) [reply tsconfig.common.json:24]: correct

Comment (gbockus-sf) [reply tsconfig.common.json:24]: it defaults to true

File: packages/salesforcedx-apex-replay-debugger/package.json (hunk 9 above)
Comment (gbockus-sf) [reply packages/salesforcedx-apex-replay-debugger/package.json:47]: 🥳

File: packages/salesforcedx-vscode-apex-replay-debugger/package.json (hunk 7 above)
Comment (RitamAgrawal) [reply packages/salesforcedx-vscode-apex-replay-debugger/package.json:31]: `salesforcedx-vscode-apex-replay-debugger` and `salesforcedx-vscode-apex-debugger` are the adapter modules and `salesforcedx-apex-replay-debugger` and `salesforcedx-apex-debugger` are the debuggers referred in this PR.

Comment (RitamAgrawal) [reply packages/salesforcedx-vscode-apex-replay-debugger/package.json:31]: And the debuggers should call `salesforcedx-utils` and not the `salesforcedx-utils-vscode`

File: packages/salesforcedx-apex-replay-debugger/package.json [hunk 10]
```diff
@@ -13,25 +13,28 @@
     "Debuggers"
   ],
   "dependencies": {
-    "@salesforce/salesforcedx-utils-vscode": "55.11.0",
+    "@salesforce/salesforcedx-utils": "55.11.0",
```
Comment (RitamAgrawal) [inline packages/salesforcedx-apex-replay-debugger/package.json:16]: The debugger module should depend on `salesforcedx-utils`.

**Review by RitamAgrawal — APPROVED** [review]

**Review by jeffb-sfdc — APPROVED** [review]

**Pull request closed without merging**